assets/js/googlesitekit/datastore/site/notifications.js

```diff
@@ -16,18 +16,111 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
 /**
  * Internal dependencies
  */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
 import { CORE_SITE } from './constants';
 import { createNotificationsStore } from '../../data/create-notifications-store';
+import { createFetchStore } from '../../data/create-fetch-store';
+import { createValidatedAction } from '../../data/utils';
 
-const notifications = createNotificationsStore(
-	'core',
-	'site',
-	'notifications',
-	{
+const ACCEPTED = 'accepted';
+const DISMISSED = 'dismissed';
+
+const isValidNotificationID = ( notificationID ) =>
+	'string' === typeof notificationID;
+
+const fetchMarkNotificationStore = createFetchStore( {
+	baseName: 'markNotification',
+	controlCallback: ( { notificationID, notificationState } ) => {
+		return API.set( 'core', 'site', 'mark-notification', {
+			notificationID,
+			notificationState,
+		} );
+	},
+	argsToParams: ( { notificationID, notificationState } ) => {
+		return { notificationID, notificationState };
+	},
+	validateParams: ( { notificationID, notificationState } = {} ) => {
+		invariant(
+			[ ACCEPTED, DISMISSED ].includes( notificationState ),
+			'notificationState must be accepted or dismissed.'
+		);
+		invariant(
+			isValidNotificationID( notificationID ),
+			'a valid notification ID is required to mark a notification.'
+		);
+	},
+} );
+
+const baseActions = {
+	/**
+	 * Marks the given notification as accepted.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} id Notification ID.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	acceptNotification: createValidatedAction(
+		( notificationID ) => {
+			invariant(
+				isValidNotificationID( notificationID ),
+				'a valid notification ID is required to accept a notification.'
+			);
+		},
+		function* ( notificationID ) {
+			const {
+				response,
+				error,
+			} = yield fetchMarkNotificationStore.actions.fetchMarkNotification(
+				{ notificationID, notificationState: ACCEPTED }
+			);
+			return { response, error };
+		}
+	),
+
+	/**
+	 * Marks the given notification as dismissed.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} id Notification ID.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	dismissNotification: createValidatedAction(
+		( notificationID ) => {
+			invariant(
+				isValidNotificationID( notificationID ),
+				'a valid notification ID is required to dismiss a notification.'
+			);
+		},
+		function* ( notificationID ) {
+			const {
+				response,
+				error,
+			} = yield fetchMarkNotificationStore.actions.fetchMarkNotification(
+				{ notificationID, notificationState: DISMISSED }
+			);
+			return { response, error };
```

Comment:
Same here, please, use the `receiveError` function to set the error for the `dismissNotifcation` action if the error is not empty.

Reply:
https://github.com/google/site-kit-wp/pull/4349/commits/b514fb6c216186b3066c316075d28a3cb5373e44